### ISSUE TITLE
SV_Position must be float4

### ISSUE THREAD

**Issue by michalo2882** [Microsoft/DirectXShaderCompiler#89]
Using min16float4 type as SV_Position results in validation error. Legacy compiler allowed using min precision with this semantic. Is the new behavior expected?

```c++
min16float4 main(min16float4 position : SV_Position, min16float2 texcoord : Texcoord) : SV_Target
{
 min16float r = texcoord.r * 0.5;
 min16float g = texcoord.g + 0.5;
 return min16float4(r, g, 0, 1);
}
```

**Comment by marcelolr** [comment]
It seems the validator is too strict. ValidateSignatureElement is not handling F16 types except for clip and cull distance.

**Comment by tex3d** [comment]
This should be fixed by #305.
